src/lib/collab/stack-client.js

```diff
@@ -275,25 +275,35 @@ export class ServiceClient {
    */
   async getDoc(noteId) {
     const res = await this.stackClient.fetchJSON('GET', this.path(noteId))
-    const formatedDoc = this.formatDoc(res)
+    const formatedDoc = this.formatDoc(res.data)
+    this.schemaVersion = formatedDoc.schemaVersion
+    return formatedDoc
+  }
+
+  /**
+   * Get the full document for a note
+   * @param {uuid} noteId
+   */
+  getFormatedDoc(note) {
+    const formatedDoc = this.formatDoc(note)
     this.schemaVersion = formatedDoc.schemaVersion
```

Comment:
Are we sure about this sideEffect (setting `this.schemaVersion`)?

Reply:
I only follow the [getDoc](https://github.com/cozy/cozy-notes/blob/3b2fd1e6f109a3e262e6d6de179c039d546f0887/src/lib/collab/stack-client.js#L276) function but without the stackClient request